File: Backend/Travle.Model/Responses/CuratorDestinationStatRow.cs
namespace Travle.Model.Responses
{
    /// <summary>
    /// Per-destination line of the curator statistics screen: how each of the curator's own submissions is
    /// performing. Engagement figures (<see cref="Views"/>, <see cref="Favorites"/>, <see cref="AverageRating"/>)
    /// apply to any status; <see cref="Bookings"/>/<see cref="Travelers"/> count the demand on tours that visit
    /// this destination (0 for a submission not yet on any booked tour). <see cref="AverageRating"/> is 0 when
    /// the destination has no (non-removed) reviews yet.
    /// </summary>
    public class CuratorDestinationStatRow
    {
        public string DestinationName { get; set; } = string.Empty;

        /// <summary>Moderation status name ("Pending" / "Approved" / "Rejected").</summary>
        public string Status { get; set; } = string.Empty;

        public int Views { get; set; }
        public int Favorites { get; set; }
        public double AverageRating { get; set; }
        public int ReviewCount { get; set; }

        /// <summary>Bookings on tours that visit this destination (this row only; may overlap other rows).</summary>
        public int Bookings { get; set; }

        /// <summary>Travelers on those bookings (sum of party sizes).</summary>
        public int Travelers { get; set; }
    }
}

File: UI/travle_core/lib/src/models/curator_destination_stat_row.dart
import 'package:json_annotation/json_annotation.dart';

part 'curator_destination_stat_row.g.dart';

/// Per-destination line of the curator statistics screen (mirrors the backend
/// `CuratorDestinationStatRow`). Engagement figures apply to any status; [bookings]
/// / [travelers] count the demand on tours that visit this destination.
/// [averageRating] is 0 when the destination has no reviews yet.
@JsonSerializable()
class CuratorDestinationStatRow {
  CuratorDestinationStatRow({
    required this.destinationName,
    required this.status,
    required this.views,
    required this.favorites,
    required this.averageRating,
    required this.reviewCount,
    required this.bookings,
    required this.travelers,
  });

  final String destinationName;

  /// Moderation status name ("Pending" / "Approved" / "Rejected").
  final String status;
  final int views;
  final int favorites;
  final double averageRating;
  final int reviewCount;
  final int bookings;
  final int travelers;

  factory CuratorDestinationStatRow.fromJson(Map<String, dynamic> json) =>
      _$CuratorDestinationStatRowFromJson(json);

  Map<String, dynamic> toJson() => _$CuratorDestinationStatRowToJson(this);
}
